packages/@react-spectrum/s2/src/DateRangePicker.tsx

```diff
@@ -13,61 +13,167 @@
-              {(date) => <CalendarCell date={date} />}
-            </CalendarGrid>
-          </RangeCalendar>
-        </Dialog>
-      </Popover>
+    <AriaDateRangePicker
+      ref={ref}
+      isRequired={isRequired}
+      {...dateFieldProps}
+      style={UNSAFE_style}
+      className={(UNSAFE_className || '') + style(field(), getAllowedOverrides())({
+        isInForm: !!formContext,
+        labelPosition,
+        size
+      }, styles)}>
+      {({isDisabled, isInvalid, isOpen, state}) => {
+        let placeholder: DateValue | undefined = placeholderValue || undefined;
+        let timePlaceholder = placeholder && 'hour' in placeholder ? placeholder : undefined;
+        let timeMinValue = props.minValue && 'hour' in props.minValue ? props.minValue : undefined;
+        let timeMaxValue = props.maxValue && 'hour' in props.maxValue ? props.maxValue : undefined;
+        let timeGranularity = state.granularity === 'hour'
+          || state.granularity === 'minute'
+          || state.granularity === 'second'
+            ? state.granularity : undefined;
+        let showTimeField = !!timeGranularity;
+        return (
+          <>
+            <FieldLabel
+              isDisabled={isDisabled}
+              isRequired={isRequired}
+              size={size}
+              labelPosition={labelPosition}
+              labelAlign={labelAlign}
+              necessityIndicator={necessityIndicator}
+              contextualHelp={contextualHelp}>
+              {label}
+            </FieldLabel>
+            <FieldGroup
+              role="presentation"
+              shouldTurnOffFocusRing={buttonHasFocus}
+              isDisabled={isDisabled}
+              isInvalid={isInvalid}
+              size={size}
+              styles={style({
+                ...fieldInput(),
+                textWrap: 'nowrap',
+                paddingStart: 'edge-to-text',
+                paddingEnd: 4
+              })({size})}>
+              <DateInputContainer>
+                <DateInput slot="start" />
+                <span aria-hidden="true" className={style({flexShrink: 0, flexGrow: 0, paddingX: 2})}>–</span>
+                <DateInput slot="end" />
+              </DateInputContainer>
+              <InvalidIndicator isInvalid={isInvalid} isDisabled={isDisabled} />
+              <div
+                className={style({
+                  flexShrink: 0,
+                  flexGrow: 1,
+                  display: 'flex',
+                  justifyContent: 'end'
+                })}>
+                <CalendarButton isOpen={isOpen} size={size} setButtonHasFocus={setButtonHasFocus} />
+              </div>
+            </FieldGroup>
+            <CalendarPopover>
+              <RangeCalendar visibleMonths={visibleMonths} createCalendar={createCalendar} firstDayOfWeek={firstDayOfWeek} />
+              {showTimeField && (
+                <div className={style({display: 'flex', gap: 16, contain: 'inline-size'})}>
+                  <TimeField
```

Comment:
This makes me wonder if RAC should have TimeField slots built in at some point.

Reply:
yeah, could definitely be nice